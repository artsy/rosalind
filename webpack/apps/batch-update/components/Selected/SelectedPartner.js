import React from 'react'
import SelectedComponent from './SelectedComponent'

const SelectedPartner = (props) => {
  return (
    <div>
      <h2>Partner</h2>
      <SelectedComponent
        stateKey='partner'
        name={props.name}
        onRemove={props.clearState}
      />
    </div>
  )
}

export { SelectedPartner }
