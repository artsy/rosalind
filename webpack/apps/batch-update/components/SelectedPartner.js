import React from 'react'

class SelectedPartner extends React.Component {
  constructor (props) {
    super(props)
    this.handleRemove = this.handleRemove.bind(this)
  }

  handleRemove (e) {
    e.preventDefault()
    this.props.onClearPartner()
  }

  render () {
    const { className, partner } = this.props
    if (partner !== null) {
      return (
        <div className={className}>
          {partner.name}
          <a href='#' className='remove'
            onClick={this.handleRemove}>âœ•</a>
        </div>
      )
    }
    return null
  }
}

SelectedPartner.propTypes = {
  partner: React.PropTypes.object.isRequired,
  onClearPartner: React.PropTypes.func.isRequired
}

/* default styled component */

import styled from 'styled-components'

const StyledSelectedPartner = styled(SelectedPartner)`
  font-weight: bold;
  margin: 0.25em 0;

  a.remove {
    display: inline-block;
    height: 2em;
    width: 2em;
    line-height: 2em;
    text-align: center;
    color: #ddd;
    margin-left: 0.5em;
    border-radius: 1em;
    text-decoration: none;
    &:hover {
      color: #333;
      cursor: pointer;
      background: #eee;
    }
  }
`

export default StyledSelectedPartner
