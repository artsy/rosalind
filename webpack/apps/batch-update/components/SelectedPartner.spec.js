import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import SelectedPartner from './SelectedPartner'

let props

beforeEach(() => {
  props = {
    partner: {
      id: 'gagosian',
      name: 'Gagosian'
    },
    onClearPartner: jest.fn()
  }
})

it('renders correctly', () => {
  const rendered = renderer.create(<SelectedPartner {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('fires the remove handler on click', () => {
  const wrapper = mount(<SelectedPartner {...props} />)
  const mockClickEvent = { preventDefault: jest.fn() }
  wrapper.find('a').simulate('click', mockClickEvent)
  expect(props.onClearPartner.mock.calls.length).toEqual(1)
})
