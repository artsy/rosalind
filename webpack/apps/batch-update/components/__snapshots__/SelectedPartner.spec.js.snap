exports[`test renders correctly 1`] = `
<div
  className="crrNnH">
  Gagosian
  <a
    className="remove"
    href="#"
    onClick={[Function]}>
    âœ•
  </a>
</div>
`;
